File: web/src/app/core/execution-result.ts
import { ErrorInfo } from "./error-info";

export class ExecutionResult<T> {
  public _success?: boolean;

  get success(): boolean {
    if (this.errors) {
      return this.errors.length === 0;
    }
    return this._success || true;
  }

  set success(value: boolean) {
    this._success = value;
  }

  constructor(public value: T, public errors?: ErrorInfo[]) { }
}

File: auth/Auth.Domain/ExecutionResult.cs
namespace Auth.Domain
{
    public class ExecutionResult
    {
        public bool Success => string.IsNullOrEmpty(Error);

        public string Error { get; set; }
    }

    public class ExecutionResult<T> : ExecutionResult where T : class
    {
        public T Result { get; set; }
    }
}
